spec/lib/submission_builder/ty2024/states/nj/documents/nj1040_spec.rb

```diff
@@ -497,6 +497,55 @@
       end
     end
 
+    describe "Other Retirement Income Exclusion - line 28a" do
+      before do
+        allow(Flipper).to receive(:enabled?).with(:show_retirement_ui).and_return(true)
+      end
+
+      context "when filer is eligible" do
+        it 'sets PensionExclusion to the line 28a value' do
+          expected = 5_000
+          allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:calculate_line_28a).and_return expected
+          expect(xml.at("PensionExclusion").text).to eq(expected.to_s)
```

Comment:
[pebble] Could we add a test for the 0 case where it does not get included in XML?

Reply:
Same with 28c